### PR TITLE
feat(outlier-detection): support for modified z-score

### PR DESCRIPTION
**Pull request by Birkbjo**
Add support for outlier detection: https://jira.dhis2.org/browse/DHIS2-11663

Did some minor refactoring to integrate this change.

### CODE REVIEW

File: src/pages/OutlierDetection/Form/ZScoreFields.js [hunk 1]
```diff
@@ -72,10 +73,15 @@ const ZScoreFields = ({
                     onChange={onOrderByChange}
                     value={orderBy}
                 >
-                    <MenuItem value={Z_SCORE} primaryText="Z-score" />
+                    <MenuItem
+                        /* API uses Z-Score for both modified and regular Z-score
+                        But we show the algorithm selected to user */
+                        value={Z_SCORE}
+                        primaryText={i18n.t(ALGORITHM_TO_LABEL_MAP[algorithm])}
```
[the remark this answers is not in the document]
Comment (Birkbjo) [reply src/pages/OutlierDetection/Form/ZScoreFields.js]: Good catch, fixed!

File: src/pages/OutlierDetection/Form/Form.js [hunk 2]
```diff
@@ -82,10 +82,15 @@ const Form = ({
                     onChange={onAlgorithmChange}
                     value={algorithm}
                 >
-                    <MenuItem value={Z_SCORE} primaryText="Z-score" />
-                    <MenuItem value={MIN_MAX} primaryText="Min-max values" />
+                    {Object.keys(ALGORITHM_TO_LABEL_MAP).map(algo => (
+                        <MenuItem
+                            key={algo}
+                            value={algo}
+                            primaryText={i18n.t(ALGORITHM_TO_LABEL_MAP[algo])}
```
Comment (mediremi) [inline src/pages/OutlierDetection/Form/Form.js:89]:
```suggestion
                            primaryText={ALGORITHM_TO_LABEL_MAP[algo]}
```

File: src/pages/OutlierDetection/OutlierAnalysisTable/OutlierAnalysisTable.js [hunk 3]
```diff
@@ -65,7 +65,7 @@ const OutlierAnalyisTable = ({
                             <TableHeaderColumn
                                 className={cssPageStyles.numericalRow}
                             >
-                                {i18n.t('Z-Score')}
+                                {i18n.t(ALGORITHM_TO_LABEL_MAP[algorithm])}
```
Comment (mediremi) [inline src/pages/OutlierDetection/OutlierAnalysisTable/OutlierAnalysisTable.js:68]:
```suggestion
                                {ALGORITHM_TO_LABEL_MAP[algorithm]}
```

**Comment by mediremi** [comment]
As the modified z-score response does not have the `mean` field, we need to update `src/pages/OutlierDetection/OutlierAnalysisTable/OutlierAnalysisTable.js` like so:

```jsx
{isZScoreAlgorithm && (
    <TableHeaderColumn
        className={cssPageStyles.numericalRow}
    >
        {algorithm === 'MOD_Z_SCORE' ? i18n.t('Median') : i18n.t('Mean')}
    </TableHeaderColumn>
```

and `src/pages/OutlierDetection/OutlierAnalysisTable/OutlierAnalysisTable/ElementRow.js` like so:

```jsx
<FormattedNumber>
  value={element.mean || element.median}
</FormattedNumber>
```

**Review by mediremi — APPROVED** [review]

**Pull request merged**